### ISSUE TITLE
Assert's error message is eagerly evaluated

### ISSUE THREAD

**Issue by fintara** [p-org/P#671]
Consider the following example:

```
var xs: map[string, int];
assert !("key" in xs), format ("xs[key] = {0}", xs["key"]);
```

If the assert fails, then the key is _in_ the map, so it can be shown in the error message. However, currently (2.0.24) this fails at runtime with:

```
<ExceptionLog> MyMachine(12) running action 'Anon_15' in state 'MyState' threw exception 'KeyNotFoundException'.
<ErrorLog> Exception 'System.Collections.Generic.KeyNotFoundException' was thrown in MyMachine(12) (state 'MyState', action 'Anon_15'): The given key 'key' was not present in the dictionary.
from location 'System.Private.CoreLib':
The stack trace is:
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Plang.CSharpRuntime.Values.PrtMap.get_Item(IPrtValue key) in /home/runner/work/P/P/Src/PRuntimes/PCSharpRuntime/Values/PrtMap.cs:line 114
   at PImplementation.MyMachine.myFun(PrtSet ensured_1) in /Users/fintara/Developer/p-model/PGenerated/CSharp/MyProject.cs:line 4279
   at PImplementation.MyMachine.Anon_15(Event currentMachine_dequeuedEvent) in /Users/fintara/Developer/p-model/PGenerated/CSharp/MyProject.cs:line 2996
   at PChecker.Actors.Actor.InvokeActionAsync(CachedDelegate cachedAction, Event e) in /home/runner/work/P/P/Src/PChecker/CheckerCore/Actors/Actor.cs:line 515
```

**Comment by ankushdesai** [comment]
Thanks for pointing it out. I will fix this in the next release.

**Comment by ankushdesai** [comment]
Thanks again for pointing this out and it is fixed in the latest release 2.0.31
